nativelib/src/main/resources/scala-native/delimcc.c

```diff
@@ -0,0 +1,355 @@
+#include "delimcc.h"
+#include <stddef.h>
+#include <stdio.h>
+#include <assert.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdatomic.h>
+
+// Defined symbols here:
+// - ASM_JMPBUF_SIZE: The size of the jmpbuf, should be a constant defined in
+// `setjmp.S`.
+// - JMPBUF_STACK_POINTER_OFFSET: The offset within the jmpbuf where
+//   the stack pointer is located. Should be defined in `setjmp.S`.
+#if defined(__aarch64__)
+#define ASM_JMPBUF_SIZE 192
+#define JMPBUF_STACK_POINTER_OFFSET (104 / 8)
+#elif defined(__x86_64__) && defined(__linux__)
```

Comment:
I was playing around with this change on my x86_64 Mac, and it works fine so far with the following change 👍 

```suggestion
#elif defined(__x86_64__) && (defined(__linux__) || defined(__APPLE__))
```

Actually, it seems the calling convention for Mac AMD64 is also System V ABI AMD64. https://developer.apple.com/library/archive/documentation/DeveloperTools/Conceptual/LowLevelABI/140-x86-64_Function_Calling_Conventions/x86_64.html (while windows x64 has a difference)

and koka's setjmp/longjmp implementation (probably copied from here?) doesn't have difference between AMD64 Linux / Mac https://github.com/koka-lang/libhandler/blob/0f1e367ca9be736b09b62480ef0039b0b3d1ad8f/src/asm/setjmp_amd64.s

I'll keep play around with AMD64 Mac, and let you know if I find something 😄

Reply:
Yes, the `setjmp`/`longjmp` implementation is indeed copied from libhandler :D